src/ar/mangaswat/src/eu/kanade/tachiyomi/extension/ar/mangaswat/MangaSwat.kt

```diff
@@ -95,108 +103,91 @@ class MangaSwat :
-    override val seriesArtistSelector = "span:contains(الناشر) i"
-    override val seriesAuthorSelector = "span:contains(المؤلف) i"
-    override val seriesGenreSelector = "span:contains(التصنيف) a, .mgen a"
-    override val seriesTypeSelector = "span:contains(النوع) a"
-    override val seriesStatusSelector = "span:contains(الحالة)"
+    override fun chapterListParse(response: Response): List<SChapter> {
+        var data = response.parseAs<ChapterListResponse>()
+        val chapters = mutableListOf<SChapter>()
+        chapters.addAll(data.results.map { it.toSChapter() })
+
+        var nextPage = data.next
+        while (nextPage != null) {
+            val nextResponse = client.newCall(GET(nextPage, response.request.headers)).execute()
+            data = nextResponse.parseAs<ChapterListResponse>()
+            chapters.addAll(data.results.map { it.toSChapter() })
+            nextPage = data.next
+        }
 
-    override fun pageListParse(document: Document): List<Page> {
-        val scriptContent = document.selectFirst("script:containsData(ts_reader)")!!.data()
-        val jsonString = scriptContent.substringAfter("ts_reader.run(").substringBefore(");")
-        val tsReader = json.decodeFromString<TSReader>(jsonString)
-        val imageUrls = tsReader.sources.firstOrNull()?.images ?: return emptyList()
-        return imageUrls.mapIndexed { index, imageUrl -> Page(index, document.location(), imageUrl) }
+        return chapters
     }
 
-    override fun chapterListSelector() = "div.bxcl li, ul div:has(span.lchx)"
+    override fun pageListRequest(chapter: SChapter): Request {
+        val id = chapter.url.removePrefix("/chapters/").substringBefore("/")
+        return GET("$apiBaseUrl/chapters/$id/", apiHeaders)
+    }
 
-    override fun chapterFromElement(element: Element) = SChapter.create().apply {
-        val urlElements = element.select("a")
-        setUrlWithoutDomain(urlElements.attr("href"))
-        name = element.select(".lch a, .chapternum").text().ifBlank { urlElements.last()!!.text() }
-        date_upload = element.selectFirst(".chapter-date")?.text().parseChapterDate()
+    override fun pageListParse(response: Response): List<Page> {
+        val chapter = response.parseAs<PageListResponse>()
+        return chapter.images.map {
+            Page(it.order, imageUrl = it.image)
+        }
     }
 
-    @Serializable
-    class TSReader(
-        val sources: List<ReaderImageSource>,
-    )
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotBlank()) {
+            return GET("$apiBaseUrl/series/?search=$query&page=$page", apiHeaders)
+        }
+
+        return super.searchMangaRequest(page, query, filters)
+    }
 
-    @Serializable
-    class ReaderImageSource(
-        val images: List<String>,
-    )
+    override fun searchMangaParse(response: Response): MangasPage {
+        if ("appswat.com" !in response.request.url.host) {
```

Comment:
Since they do change domains often this can easily be missed. Compare against `baseUrl` instead. Convert it to a `HttpUrl` first.